frontends/PyCDE/src/pycde/system.py

```diff
@@ -101,6 +101,11 @@ def create_physical_region(self, name: str = None):
       physical_region = PhysicalRegion(name)
     return physical_region
 
+  def create_entity_extern(self, tag: str, metadata=""):
+    with self._get_ip():
+      entity_extern = circt.dialects.msft.EntityExternOp.create(tag, metadata)
+    return entity_extern
```

Comment:
We shouldn't ever return a raw op since we need to manage them ourselves to ensure they get destroyed/invalidated at the proper time. I would suggest just returning the symbol (which might need to be renamed). I have some facilities in System which manage symbols at the top level.

Reply:
I'm not sure I follow. If we returned a python object that wrapped a reference to the op, wouldn't it be the same? This op should get destroyed with the mlir::ModuleOp, just like anything else pycde creates. Wouldn't the python reference be valid till the op is removed in msft-to-hw either way?

Reply:
Regardless of lifetimes, all the similar APIs return a wrapper in pycde, so we should here. I'll make that change similar to how PhysicalRegions get handed out.

Reply:
There are multiple places currently in pycde which have a python wrapper which holds a reference to a circt op... These all need to be cleaned up to have a level of indirection in `System` at some point so that python will release the reference when the passes get run. Important since the user won't necessarily exit after running the passes and may keep references to various things around.

Reply:
Having these wrapper classes makes cleaning up easier.